File: src/test/java/org/example/bankramenserver/domain/category/presentation/CategoryControllerTest.java
package org.example.bankramenserver.domain.category.presentation;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.setup.MockMvcBuilders;

import static org.hamcrest.Matchers.greaterThan;
import static org.hamcrest.Matchers.hasItem;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

class CategoryControllerTest {

    private MockMvc mockMvc;

    @BeforeEach
    void setUp() {
        mockMvc = MockMvcBuilders.standaloneSetup(new CategoryController())
                .build();
    }

    @Test
    void getCategoriesReturnsDefaultCategoryList() throws Exception {
        mockMvc.perform(get("/categories"))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$.categories.length()", greaterThan(0)))
                .andExpect(jsonPath("$.categories[*].code", hasItem("FOOD")))
                .andExpect(jsonPath("$.categories[*].displayName", hasItem("식비")))
                .andExpect(jsonPath("$.categories[*].code", hasItem("SALARY")))
                .andExpect(jsonPath("$.categories[*].displayName", hasItem("급여")));
    }
}

File: src/test/kotlin/org/example/bankramenserver/domain/category/presentation/CategoryControllerTest.kt
package org.example.bankramenserver.domain.category.presentation

import org.hamcrest.Matchers.greaterThan
import org.hamcrest.Matchers.hasItem
import org.junit.jupiter.api.Test
import org.springframework.test.web.servlet.setup.MockMvcBuilders
import org.springframework.test.web.servlet.get
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

class CategoryControllerTest {
    private val mockMvc = MockMvcBuilders.standaloneSetup(CategoryController()).build()

    @Test
    fun getCategoriesReturnsDefaultCategoryList() {
        mockMvc.get("/categories") {
            accept = org.springframework.http.MediaType.APPLICATION_JSON
        }.andExpect {
            status { isOk() }
            jsonPath("$.categories.length()", greaterThan(0))
            jsonPath("$.categories[*].code", hasItem("FOOD"))
            jsonPath("$.categories[*].displayName", hasItem("식비"))
            jsonPath("$.categories[*].code", hasItem("SALARY"))
            jsonPath("$.categories[*].displayName", hasItem("급여"))
        }
    }
}
